subsys/net/ip/net_if.c

```diff
@@ -1727,6 +1727,14 @@ void net_address_lifetime_timeout(void)
 
 static void address_start_timer(struct net_if_addr *ifaddr, uint32_t vlifetime)
 {
+	/* Make sure that we do not insert the address twice to
+	 * the lifetime timer list.
+	 */
+	if (!sys_slist_is_empty(&active_address_lifetime_timers)) {
+		sys_slist_find_and_remove(&active_address_lifetime_timers,
+					  &ifaddr->lifetime.node);
+	}
```

Comment:
Simply call `find_and_remove`, it will return on the first head check?

```suggestion
	sys_slist_find_and_remove(&active_address_lifetime_timers,
				  &ifaddr->lifetime.node);
```

Reply:
Works ok without the empty list check, dunno why I had some issues there. Anyway, I removed the check.